### ISSUE TITLE
Quadro Mosaic Profile Error

### ISSUE THREAD

**Issue by Slippery-Silvia** [terrymacdonald/DisplayMagician#30]
Hi, having trouble getting surround (mosaic) to enable with nvidia Quadro RTX 4000.

I am not sure because Quadro card doesnt specifically have nvidia surround, its called Mosaic. If this could be causing an issue?

My display profile is saved but when i try enable it. the adjusting display devices countdown closes out with 9 seconds to go and changes are not applied.

**Comment by terrymacdonald** [comment]
Hi @Slippery-Silvia,

NVIDIA Surround is just the marketing name for Mosaic. The code used is exactly for those Quadro cards as it is for the RTX GeForce models (at least as far as I'm aware). So the code *should* work with your Quadro RTX 4000.

I'll need you to create some TRACE log files for me to be able to troubleshoot your problem. DisplayMagician can generate extensive logs which will enable me to better determine where your error lies. By default DisplayMagician just generates a WARNING level log file, but to troubleshoot your error I'll need a TRACE level log file.

Could you please generate a TRACE DisplayMagician.log file using the instructions below, and upload it here for me to review? Also please upload your DisplayProfiles_1.0.json and Shortcuts_1.0.json files too (they are in the Profiles and Shortcuts subdirectories near to where you will find the DisplayMagician.log file.

To generate a DisplayMagician.log file with TRACE entries in it, please do the following:

- Press WindowsKey + R to bring up a Run prompt
- Click the Browse button
- Navigate to `C:\Program Files\DisplayMagician\DisplayMagician.exe`, and click the Open button
- Go to the end of the command, and add a space, and then add `--trace` on the end.
- The whole command should read `"C:\Program Files\DisplayMagician\DisplayMagician.exe" --trace`
- Press the Run button.

This will force DisplayMagician to start logging TRACE commands until DisplayMagician is closed down.

Now. perform the actions that don't work. In your case, please open up the Display Profile window, and then try to change to the display profile that doesn't work. Then close down DisplayMagician.

**Then straight away (before restarting DisplayMagician)** open up DisplayMagician.log file and check that it has "TRACE" commands in the log file, because then we know that the --trace option worked properly. If you don't see the word 'TRACE' in the log file then the 

Straight away, upload the DisplayMagician.log file along with the DisplayProfiles_1.0.json and Shortcuts_1.0.json files to this issue. Once I have the DisplayMagician.log file with TRACE commands in it I'll be able to see if there is anything wrong, and I'll check your two .json config files to see if there is anything odd in there too.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Oh @Slippery-Silvia, the other thing I need to know is whether you recently updated your video card driver? Sometimes the NVIDIA video driver changes the information it reports, and that affects the 'fingerprint' that we take for each of your displays. This doesn't seem to be the problem you have (based on your description) but it is something to remember for the future.

Sometimes if you recreate your Display Profile, and then edit your Game Shortcuts to use that new Display Profile it can fix the problem.

**Comment by Slippery-Silvia** [comment]
Hi @terrymacdonald, Thanks for getting back so quick. just got back from a short holiday. 

a quick look shows that my mosaic profile does not seem to re registered as surround?

2021-07-25 15:03:33.5062|DEBUG|DisplayMagician.Program|ProfileRepository/GetActiveProfile: Retrieving the currently active profile.
2021-07-25 15:03:38.0624|DEBUG|DisplayMagician.Program|Program/ApplyProfile: Mosaic profile we want to use does not use NVIDIA Surround.
2021-07-25 15:03:38.0624|DEBUG|DisplayMagician.Program|Program/ApplyProfile: Tripple profile currently in use does not use NVIDIA Surround.

Have attached the files you need in a .zip file for me detailed information.

[Quadro-DisplayMagician_Logs.zip](https://github.com/terrymacdonald/DisplayMagician/files/6873397/Quadro-DisplayMagician_Logs.zip)

**Comment by terrymacdonald** [comment]
@Slippery-Silvia I've found the issue. It looks like the Quadro range of cards have a different internal UID than the GeForce RTX range. Those use UID5120. I'll need to add the Quadro UID to the detection logic for the next release of DisplayMagician (which will be v1.1.5 in a couple of weeks).

In order to do that, I'll need your help in getting that UID. 

Can you please perform the following steps:

- Open a run prompt (Ctrl+R)
- Type 'system', and select **System Information** from the list of options shown
- Expand out the **Components** section
- Expand out the **Display** section
- Please take a screenshot or copy the text in the Display section. It should look something like this (but NVIDIA of course :)):

```
Name	AMD Radeon HD 7700 Series
PNP Device ID	PCI\VEN_1002&DEV_665C&SUBSYS_04521043&REV_00\4&2283F625&0&0019
Adapter Type	AMD Radeon Graphics Processor (0x665C), Advanced Micro Devices, Inc. compatible
Adapter Description	AMD Radeon HD 7700 Series
Adapter RAM	1.00 GB (1,073,741,824 bytes)
Installed Drivers	C:\Windows\System32\DriverStore\FileRepository\u0368456.inf_amd64_fc65705fd5034968\B367348\aticfx64.dll,C:\Windows\System32\DriverStore\FileRepository\u0368456.inf_amd64_fc65705fd5034968\B367348\aticfx64.dll,C:\Windows\System32\DriverStore\FileRepository\u0368456.inf_amd64_fc65705fd5034968\B367348\aticfx64.dll,C:\Windows\System32\DriverStore\FileRepository\u0368456.inf_amd64_fc65705fd5034968\B367348\amdxc64.dll
Driver Version	27.20.20903.8001
INF File	oem41.inf (ati2mtag_R505 section)
Colour Planes	Not Available
Colour Table Entries	4294967296
Resolution	2560 x 1440 x 60 hertz
Bits/Pixel	32
Memory Address	0xD0000000-0xDFFFFFFF
Memory Address	0xE0000000-0xE07FFFFF
I/O Port	0x0000E000-0x0000E0FF
Memory Address	0xFCE00000-0xFCE3FFFF
IRQ Channel	IRQ 4294967274
I/O Port	0x000003B0-0x000003BB
I/O Port	0x000003C0-0x000003DF
Memory Address	0xA0000-0xBFFFF
Driver	C:\WINDOWS\SYSTEM32\DRIVERSTORE\FILEREPOSITORY\U0368456.INF_AMD64_FC65705FD5034968\B367348\AMDKMDAG.SYS (27.20.20903.8001, 78.86 MB (82,687,200 bytes), 18/06/2021 2:56 am)
```

This will hopefully give me the information I need to update DisplayMagician so it will work with all Quadro cards too!

Just for your information, I'm currently rewriting the Display Profile manipulation code from scratch, so that it is a LOT more robust, and handles things like HDR as well as AMD Eyefinity (the AMD version of NVIDIA Mosaic/Surround). This problem will disappear when that code appears in DisplayMagician v2.0.0, but that's in another month or so. v2.0.0. will also be a breaking change, so will be an optional upgrade.

Good luck, and please reply with that text so I can do the update. 

Are you able to do some testing after I make the changes to make sure that the Quadro detection logic is working? It would really help me out if you could.

Thanks
Terry

**Comment by Slippery-Silvia** [comment]
@terrymacdonald Happy to help test the Quadro detection logic works. Thank you for all the hard work in fixing this problem for Quadro card users. Looking forward to v2.0.0

Name	NVIDIA Quadro RTX 4000
PNP Device ID	PCI\VEN_10DE&DEV_1EB1&SUBSYS_12A010DE&REV_A1\4&2283F625&0&0019
Adapter Type	Quadro RTX 4000, NVIDIA compatible
Adapter Description	NVIDIA Quadro RTX 4000
Adapter RAM	(1,048,576) bytes
Installed Drivers	C:\Windows\System32\DriverStore\FileRepository\nv_dispwi.inf_amd64_b52b5e8d70ad7968\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nv_dispwi.inf_amd64_b52b5e8d70ad7968\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nv_dispwi.inf_amd64_b52b5e8d70ad7968\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nv_dispwi.inf_amd64_b52b5e8d70ad7968\nvldumdx.dll
Driver Version	30.0.14.7141
INF File	oem77.inf (Section071 section)
Colour Planes	Not Available
Colour Table Entries	4294967296
Resolution	1920 x 1080 x 165 hertz
Bits/Pixel	32
Memory Address	0xFB000000-0xFBFFFFFF
Memory Address	0xD0000000-0xDFFFFFFF
Memory Address	0xE0000000-0xE1FFFFFF
I/O Port	0x0000E000-0x0000E07F
IRQ Channel	IRQ 4294967250
I/O Port	0x000003B0-0x000003BB
I/O Port	0x000003C0-0x000003DF
Memory Address	0xA0000-0xBFFFF
Driver	C:\WINDOWS\SYSTEM32\DRIVERSTORE\FILEREPOSITORY\NV_DISPWI.INF_AMD64_B52B5E8D70AD7968\NVLDDMKM.SYS (30.0.14.7141, 35.69 MB (37,420,672 bytes), 20/07/2021 5:57 PM)

**Comment by terrymacdonald** [comment]
@Slippery-Silvia Some great news. I think I've fixed the faulty logic. I'd love if you could test the DisplayMagician v1.2.0 that I've included here. Please install it and then:

- Delete the old Mosiac profile you created earlier
- Setup Surround/Mosiac again with the NVIDIA display setup
- Go back to DisplayMagician and save the new Mosiac profile.
- See if you can swap between your Surround/Mosiac and non-surround display profiles. 

If everything goes to plan, it'll work. If not, you'll hopefully have some errors you can send me with another TRACE DisplayMagician.log file :)

[DisplayMagicianSetup-v1.2.0-test.zip](https://github.com/terrymacdonald/DisplayMagician/files/6876713/DisplayMagicianSetup-v1.2.0-test.zip)

**Comment by Slippery-Silvia** [comment]
Hi @terrymacdonald, Unfortunately getting the same error again.

Updated to v1.2.0, deleted old profiles and created new profiles. Same error closing out halfway through countdown.
Uninstalled, restarted, reinstalled, created new profiles. Same error again.

New log file to review attached.

[DisplayMagician.log](https://github.com/terrymacdonald/DisplayMagician/files/6877365/DisplayMagician.log)

**Comment by terrymacdonald** [comment]
Thanks for those last logs @Slippery-Silvia. 

Are you please able to provide a screen shot of the way that you set up the Quadro Mosiac screen layout for me please? I don't have a Quadro card, so I can't do it myself. If you could take a screenshot and provide that to me then I will be able to see how that is different from the **NVIDIA Control Panel** way that the RTX GeForce cards use. I've included a screenshot of how I do it for comparison.

![NVIDIA Set Up Surround window](https://user-images.githubusercontent.com/5721772/127065115-f8451496-c710-4548-bc32-2166ebdd800a.png)

**Unfortunately, those last logs confirmed that I won't be able to add Mosaic support to the v1.x line of DisplayMagician versions that use the current NVIDIA display library.** It looks as though Mosaic is handled differently from NVIDIA Surround as NVIDIA Surround changes the name of the single display to "NVS Surround", and DisplayMagician v1.x uses that fact as part of detection routine for identifying that Surround exists. It seems from the logs that the Mosiac keeps the name of one of the Displays and just makes it really big (5780 x1080 rather than 1920x1080 if the DisplayProfiles_1.0.log). 

Instead, this will be fixed in the upcoming DisplayMagician v2.0.0. I decided about a month and a half ago that the DisplayMagician video libraries needed recreating, as they made some assumptions that meant they missed some of the nuances of display setups that people had. Things like:

- Bezel settings
- HDR settings
- Which display is the primary
- AMD Eyefinity

and other things. So I decided to rewrite the NVIDIA, AMD and Windows CCD libraries again from scratch, to cover all these scenarios. 

My plan for the NVIDIA library is to do the detection based on the combination of the PCI Vendor ID of the video card, in conjunction with the NVAPI driver, so that I'll be able to identify NVIDIA hardware and software and look for them. This should be able to identify and configure the display profiles for Mosaic setups as well as Surround, but I will need your help to test.

I'm currently developing a test program called NVIDIAInfo which contains the NVIDIA library I'm writing. I decided to do this in a separate program so that I can get people to test the display profile code without having to wire it into DisplayMagician. 

Are you happy to help me do some testing with NVIDIAInfo when it's ready so that we can eventually add Quadro card support to DisplayMagician? It will be a 2-3 weeks before I'll have the library completed enough to use, but it'll be very helpful if you'd help me test.  Eventually, when the Quadro support works in NVIDIAInfo, you'll be able to use NVIDIAInfo to change your display profiles while you wait for DisplayMagician v2.0.0 to be finished :). 

Thanks
Terry

**Comment by Slippery-Silvia** [comment]
@terrymacdonald 

Am happy to help out with any testing. 

Some screen shots of the mosaic setup process
![Capture2](https://user-images.githubusercontent.com/51182742/127071960-adc9b1be-bae6-48ab-9517-0cbcfd6c01d0.PNG)
![Capture3](https://user-images.githubusercontent.com/51182742/127071967-afdd94e6-8c94-48ca-b2da-0c127dd6c714.PNG)
![Capture4](https://user-images.githubusercontent.com/51182742/127071971-23ddcccd-a593-460b-85e1-23b4d2e79b50.PNG)
![Capture5](https://user-images.githubusercontent.com/51182742/127071976-7adc7ff0-fa78-46d2-902c-5f3def6f3d20.PNG)
![Capture](https://user-images.githubusercontent.com/51182742/127071980-82b73f4d-eb3a-43f4-b240-9bdbcd047853.PNG)

**Comment by terrymacdonald** [comment]
@Slippery-Silvia I've just finished the first 'basic' version of NVIDIAInfo v0.0.1 Alpha. Can you please test it for me to see whether it works with a Quadro Mosaic setup? I've written a new NVIDIA library from scratch, and this is the first 'mostly' working version. It will successfully save and load NVIDIA surround/mosaic settings, but it won't do NVIDIA HDR or anything else. It is a pre-release version of the software, but I need you to do some testing so I can be sure its working.

Please download the NVIDIAInfo v0.0.1 zip file from https://github.com/terrymacdonald/NVIDIAInfo/releases/tag/v0.0.1, unzip the zip file and then follow the instructions in the release notes to save 2 different display profiles - one using Mosaic, and one not using Mosaic. Then please try to swap between them using the load command.

Once you've finished testing, please post your NVIDIAInfo.log files here for me to see, and tell me how the testing went.

Thanks
Terry

**Comment by Slippery-Silvia** [comment]
Hi @terrymacdonald 

Sorry i am not a developer or anything so my feedback may be a bit limited. 

I have downloaded and NVDIAInfo-v0.0.1. Am i meant to run the NVIDIAInfo.exe ? if so when i try to do so it just closes right away.

Am i meant to use NVIDIAInfo to save "monitorarrangement".cfg files? when i try run the command in cmd it is not recognised.

Log file attached.

[NVIDIAInfo.log](https://github.com/terrymacdonald/DisplayMagician/files/6991591/NVIDIAInfo.log)

**Comment by Slippery-Silvia** [comment]
I am unable now to save new display profiles using display magician

![image](https://user-images.githubusercontent.com/51182742/129551725-b3b46cd7-b745-4e99-8363-2bdf03147001.png)

**Comment by Slippery-Silvia** [comment]
Mate @terrymacdonald 

Don't worry about that last message. I figured out how to run it.

Works good as gold! awesome stuff, very quick too! much quicker than display magician.

Used the load command to keep changing back between the different profiles and no problems so far. 

New log file to review [NVIDIAInfo.log](https://github.com/terrymacdonald/DisplayMagician/files/6991940/NVIDIAInfo.log)

**Comment by Slippery-Silvia** [comment]
Also let you know. 

I created a "mycooldisplay".cmd file with "NVIDIAInfo load mycooldisplay.cfg"

when i open that .cmd file it will load the display configuration. as long as it is saved in the same folder as NVIDIAInfo.exe and the cfg folder is stored will work.

have it working with my stream deck, set a button to open the .cmd file and now can change profiles with a simple button. 
little video https://photos.app.goo.gl/ampeyA4tmUJHsYHD9
![image](https://user-images.githubusercontent.com/51182742/129561469-7c801ff7-5280-45c5-8e98-6cd56779a82e.png)

**Comment by terrymacdonald** [comment]
Excellent news! I was hoping that was the case. Thank you for confirming.

Ok, so the next steps are that I need to add the NVIDIA specific HDR
settings into the config file, and then the step after that is adding the
print display command, and the displayids command. Once that's completed,
the first proper version of NVIDIAInfo will be done, and I'll ask you to
download it and test it again.

Once NVIDIAInfo is tested and that I know the NVIDIA display manipulation
library works, I will then port it over to DisplayMagician v2.0.0. At that
point, you should be able to use DisplayMagician as everyone else does, and
I will be able to close this issue!

Thanks again for testing. It's only with help from people like you that I
can make improvements like this.

Terry

**Comment by terrymacdonald** [comment]
@Slippery-Silvia I've added the ability to save NVIDIA HDR information to NVIDIAInfo v1.0.0, as well as checking that the display configuration is available before attempting to apply it. I've tested it successfully internally, but now I need you to test it again for me to make sure I haven't affected the Quadro card support.

**IMPORTANT - The configuration file format has changed so you will need to create new cfg files for this as the old configs will not work with NVIDIA v1.0.0.**

So, can you please download `NVIDIA v1.0.0.zip` from https://github.com/terrymacdonald/NVIDIAInfo/releases/tag/v1.0.0, and then test that it works?

Please create four new configuration files for the following scenarios:

- A config file for a Mosaic screen with all 3 screens joined as a single screen
- A config file for a Triple screen with the left screen as the main screen
- A config file for a Triple screen with the middle screen as the main screen
- A config file for only two screens with the right screen turned off

Please test swapping between those 4 different configs to make sure that it works correctly transitioning between each configuration.

If it all works without any issues, then I guess I'm moving on to integrating this library with DisplayMagician!

**Comment by Slippery-Silvia** [comment]
Hi @terrymacdonald 

All working perfectly. Tested configs 3L, 3M, 3R & Mosaic.

I am not sure how to test turning a single screen off?
I have multi monitor tool installed which i used to turn on/off a 4th screen or change from either sim racing to workstation setup. Turning a single screen off with multimonitortool and then saving an nvidiainfo cfg did not work for me. 

For me whenever i turn a screen off into standby mode it still shows in windows. its not until i restart or unplug the cable the monitor will not register.

Not sure if you need log file, here it is if so.
[NVIDIAInfo.log](https://github.com/terrymacdonald/DisplayMagician/files/7026822/NVIDIAInfo.log)

**Comment by terrymacdonald** [comment]
Great news!

Don't worry about the turning off of the other screen. I was meaning
'disconnecting' it from within windows display setup, but it doesn't
matter. It was the setting of the primary display I wanted to check, and
that's worked on.

I've started to integrate the new libraries into DisplayMagician v2.0.0 now
that I know they're working. Thanks for your help again!

Terry

**Comment by Slippery-Silvia** [comment]
Awesome! Glad I could be helpful. 

Thank you and good luck with implementing it into 2.0. can't wait for it's release.

**Comment by terrymacdonald** [comment]
And another update. This has been a monster challenge, but I think I'm nearing the end of the NVIDIA chapter at least. Things that the new DisplayMagician can do:

Swap between surround and non-surround profiles within about 14 seconds
Handle surround and non-surround screens together
Works with NVIDIA mosaic cards now
Uses the PCI video card vendor ID to recognise NVIDIA and AMD vendors, and sets it's driver mode accordingly
Can be 'forced' to use a different video card mode if needed (unlikely to be used, but hey, why not)
Can set/clear a Desktop Background per profile if you want
Stores and applies Windows HDR and NVIDIA HDR settings
Records all settings provided by the NVIDIA driver and windows drivers under Windows 10, meaning that you display refresh settings are stored and set, as are bezel distances, or any other weird and wonderful settings you creative users are able to configure :)
But at present there are still bits I have to finish before I can craft any sort of production ready release:

It is very slow to load. I've identified this is due to DisplayMagician keeping on calling the video card libraries. I need to implement some caching to ensure that the video detection is only run at programme launch and when screen layout changes in some way.
I need to add AMD library support. This is the core reason why I started this journey, so it seems a little disingenuous to release DisplayMagician without this!
I need to add a nice message that popups when an old DisplayMagician file is detected, that informs the user that this is a new version of DisplayMagician that uses a brand new DisplayProfile format that will require them to create a brand new set of Display Profile layouts.
I need to test, and to provide my beta testers an early release of the software so that I iron out any bugs before a general release.
I need to add the ability to test the upgrade mechanism without upgrading the production upgrade functionality. This is needed so I can iron out any issues before a general release. Very important as DisplayMagician now has over 1500 users!
And I think that's about it!

Some cool things to come for sure, and it's taken a lot of hard work to get this close. Another little while and we're hopefully good to go!

Thanks
Terry

**Comment by terrymacdonald** [comment]
@Slippery-Silvia,

I finally have an test version of DisplayMagician v2.0.0 to share, and I'd be honoured if you'd do some testing for me.

I've attached DisplayMagician v2.0.0-alpha1.msi to this issue, and I'm really interesting to find out if it works on your system. This version of DisplayMagician uses two completely new display manipulation libraries that I created (one for NVIDIA and one for Windows CCD) so it needs some proper testing. Please feel free to test complicated setups like 2 screens in a Surround screen with another two screens next to it, or with different refresh rates, or HDR on only some of the screens. **I really want to know if it will break.**

IMPORTANT: DisplayMagician v2.0.0 uses a completely new Display Profile format. This means that you will need to create brand new Display Profiles when you first start DisplayMagician v2.0.0, and you will then need to edit your Game Shortcuts to use those new Display Profiles.

**How to install this version of DisplayMagician:**

- Download and run the DisplayMagicianSetup-v2.0.0-alpha1.msi installer
- Once installed, run DisplayMagician v2.0.0
- Go straight to 'Settings' on the main screen
- Change the Logging Level to 'Full application trace'
- Click 'Back'
- Click the 'Display Profiles' button
- Setup the Display Profiles that you want to use (e.g. open up Windows Setup or NVIDIA Setup and change your display configuration until it's the way you want it, then swap back to DisplayMagician and save the DisplayProfile).
- Edit your Game Shortcuts to use the new Display Profiles you just set up.

Then now please start using DisplayMagician v2.0.0 and see what breaks!

If you find a bug, please log it as a new issue at https://github.com/terrymacdonald/DisplayMagician/issues/new/choose

Thanks
Terry

[DisplayMagicianSetup-v2.0.0-alpha1.zip](https://github.com/terrymacdonald/DisplayMagician/files/7139878/DisplayMagicianSetup-v2.0.0-alpha1.zip)

**Comment by terrymacdonald** [comment]
One thing I forgot to mention is that it currently won't work on any system that has an AMD video card. I've not finished the AMD library, yet AMD detection is already plumbed into DisplayMagician, which means that DisplayMagician will attempt to use the non-existent AMD library if an AMD video card is detected, and it will just crash.

Thanks
Terry

**Comment by Slippery-Silvia** [comment]
Hi @terrymacdonald 

Thank you for the update, so far everything is working perfectly fine for me :D 

Liking the snappiness of the interchange, much faster than before.

I did get an error trying to install said missing .dll or something but i clicked ignore on the message and it seems to be fine.

[DisplayMagician.log](https://github.com/terrymacdonald/DisplayMagician/files/7143357/DisplayMagician.log)

**Comment by terrymacdonald** [comment]
@Slippery-Silvia Great news. I've got another DisplayMagican v2 build to test. This one has had a lot of local testing, and I've fixed a lot of bugs that I've found. And now I think it's ready for another round of testing. 

I'd really love it if you could test this out for me. To install DisplayMagician v2.0.0-alpha2:

- Firstly, uninstall DisplayMagician v2.0.0 alpha1 if you installed it earlier.
- Secondly, download and install DisplayMagician v2.0.0 alpha2 from this issue
- Thirdly, test it out :).

Hopefully it goes well for you and does everything it should, but if you do find any issues, please log them as new issues here: https://github.com/terrymacdonald/DisplayMagician/issues/new?assignees=terrymacdonald&labels=bug&template=bug_report.md&title=

Thanks 
Terry

[DisplayMagicianSetup-v2.0.0-alpha2.zip](https://github.com/terrymacdonald/DisplayMagician/files/7229458/DisplayMagicianSetup-v2.0.0-alpha2.zip)

**Comment by terrymacdonald** [comment]
Great news! DisplayMagician v2.0.1 has been released, and it should fix this very issue. Please test it out, and if you still have any issues please log a new issue. You can get DisplayMagician v2.0.1 from here: https://github.com/terrymacdonald/DisplayMagician/releases/tag/v2.0.1

I'm closing this issue as the fix is in DisplayMagician v2.0.1, and I need to clear out the completed issues so I am able to work on the new issues that come in.

Thanks
Terry
